groovy/postInit/progression/Batteries.groovy

```diff
@@ -10,50 +10,129 @@ import gregtech.api.unification.material.Materials;
+  metaitem('drum.lead').noreturn()
 ]);
 
 mods.gregtech.assembler.recipeBuilder()
-        .inputs(ore('stickLongLead') * 2)
-        .inputs(ore('plateLead') * 4)
-        .outputs(metaitem('drum.lead'))
-        .duration(200)
-        .EUt(16)
-        .notConsumable(Globals.circuit(2))
-        .buildAndRegister()
+  .inputs(ore('stickLongLead') * 2)
+  .inputs(ore('plateLead') * 4)
+  .outputs(metaitem('drum.lead'))
+  .duration(200)
+  .EUt(16)
+  .notConsumable(Globals.circuit(2))
+  .buildAndRegister()
 
 MIXER_RECIPES.recipeBuilder()
-.fluidInputs(Materials.SulfurTrioxide.getFluid(1000))
-.fluidInputs(Materials.Water.getFluid(1000))
-.fluidOutputs(Materials.SulfuricAcid.getFluid(1000))
-.EUt(7)
-.duration(40)
-.buildAndRegister()
+  .fluidInputs(Materials.SulfurTrioxide.getFluid(1000))
+  .fluidInputs(Materials.Water.getFluid(1000))
+  .fluidOutputs(Materials.SulfuricAcid.getFluid(1000))
+  .EUt(7)
+  .duration(40)
+  .buildAndRegister()
+
+def makeBattery(cathode, annode, seperator, electrolyte, outputMaterial, tierRanges, tierOffset) {
+  for(tier in tierRanges) {
+    // Can recipie
+    def tierToKey = Globals.voltageTiers.indexOf(tier)
+    def tierToKeyOffset = tierToKey - tierOffset
+    def myRecipie = mods.gregtech.assembler.recipeBuilder()
+      .inputs(cathode * Math.min(Math.max(tierToKeyOffset, 1), 4))
+      .inputs(annode * Math.min(Math.max(tierToKeyOffset, 1), 4))
+      .inputs(metaitem(Globals.cableTiers[tier]) * Math.min(Math.max(tierToKeyOffset * 2, 1), 4)) // too lazy didnt add the other copper variant
+      .outputs(metaitem("battery.hull.${outputMaterial}.${tier}"))
+      .duration(20 * Math.max(tierToKeyOffset, 1))
+      .EUt(Globals.voltageTiersInt[tierToKey - 1])
+
+    if (tierToKey > 2) {
+      myRecipie.fluidInputs(fluid('polyvinyl_chloride') * (144 * Math.min(Math.max(tierToKeyOffset * 3, 1), 9)))
+    } else {
+      myRecipie.fluidInputs(fluid('plastic') * 144)
+    }
+
+    if (seperator != null) {
+      myRecipie.inputs(seperator)
+    }
+
+    myRecipie.buildAndRegister()
+
+    // Canning recipie
+    mods.gregtech.canner.recipeBuilder()
+      .inputs(metaitem("battery.hull.${outputMaterial}.${tier}"))
+      .inputs(electrolyte * Math.min(Math.max(tierToKeyOffset, 1), 4))
+      .outputs(metaitem("battery.${tier}.${outputMaterial}"))
+      .duration(10 * Math.max(tierToKeyOffset, 1)) // start incrementing at MV (2)
+      .EUt(Globals.voltageTiersInt[tierToKey - 1])
+      .buildAndRegister()
+    println("[INFO]: Magic Numbers@${tier} => CA: ${Math.min(Math.max(tierToKeyOffset, 1), 4)} | C: ${Math.min(Math.max(tierToKeyOffset * 2, 1), 4)} | D: ${Math.max(tierToKeyOffset, 1)} | V: ${Globals.voltageTiersInt[tierToKey - 1]}")
+  }
+  println("[INFO]: Built battery ${outputMaterial} at tier range of ${tierRanges}")
+}
+
+// lipf6 missing add
+makeBattery(
+  metaitem("cathode.lithium_cobalt_oxide"), 
+  metaitem("anode.graphite"), 
+  null, //ore("seperator.mv"), 
+  metaitem("lipf6"), 
```

Comment:
lipf6 should be a dust and not a metaitem, and its name is not consistent with the naming scheme we are using. Register "lithium_hexafluorophosphate" in RegisterSolids.groovy and change this line to metaitem("dustLithiumHexafluorophosphate").